serialization/serialization_config.go

```diff
@@ -23,75 +23,122 @@ import (
 
 // Config contains the serialization configuration of a Hazelcast instance.
 type Config struct {
-	// GlobalSerializer is the serializer that will be used if no other serializer is applicable.
-	GlobalSerializer Serializer `json:"-"`
-	// CustomSerializers is a map of object types and corresponding custom serializers.
-	CustomSerializers map[reflect.Type]Serializer `json:"-"`
-	// IdentifiedDataSerializableFactories is a map of factory IDs and corresponding IdentifiedDataSerializable factories.
-	IdentifiedDataSerializableFactories []IdentifiedDataSerializableFactory `json:"-"`
-	// PortableFactories is a map of factory IDs and corresponding Portable factories.
-	PortableFactories []PortableFactory `json:"-"`
-	// ClassDefinitions contains ClassDefinitions for portable structs.
-	ClassDefinitions []*ClassDefinition `json:"-"`
-	PortableVersion  int32
+	globalSerializer                    Serializer
+	customSerializers                   map[reflect.Type]Serializer
+	identifiedDataSerializableFactories []IdentifiedDataSerializableFactory
+	portableFactories                   []PortableFactory
+	classDefinitions                    []*ClassDefinition
```

Comment:
We won't be able to specify these in JSON, right?

Reply:
No, it's not possible to specify arbitrary objects in JSON. I've unexported those fields because of that.